transport/http/server.go

```diff
@@ -70,8 +70,22 @@ func ServerErrorEncoder(ee ErrorEncoder) ServerOption {
 // of error handling, including logging in more detail, should be performed in a
 // custom ServerErrorEncoder or ServerFinalizer, both of which have access to
 // the context.
+// Deprecated: Use ServerErrorHandler instead.
 func ServerErrorLogger(logger log.Logger) ServerOption {
-	return func(s *Server) { s.logger = logger }
+	return func(s *Server) {
+		if s.errorHandler == nil {
```

Comment:
Why the explicit check?

Reply:
Because the ErrorLogger should be considered deprecated and if someone uses an ErrorHandler, it should take precedence.

Reply:
IMO ServerErrorLogger should now just be a helper function that calls e.g.

```go
func ServerErrorLogger(logger log.Logger) ServerOption {
    return func(s *Server) { s.errorHandler = LoggingErrorHandler(logger) }
}
```

No need for precedence. You can keep the deprecation warning.